lib/repos/UserRepo.js

```diff
@@ -277,9 +284,14 @@ class UserRepo {
 		user.metadata.created = now;
 		user.metadata.updated = now;
 
-		const createdUser = await this._collection.insertOne(user);
+		const result = await this._collection.insertOne(user);
 
-		return new UserModel(createdUser.ops[0]);
+		if (result.acknowledged) {
+			const createdUser = await this._collection.findOne({ _id: result.insertedId });
+			return new UserModel(createdUser);
+		} else {
+			throw new Error("Insert failed");
+		}
```

Comment:
We are usually not handling exception in the repo. Hope this properly handled in the controller.